model/validation.go

```diff
@@ -669,3 +661,89 @@ func ValidateDestinationPolicy(msg proto.Message) error {
 
 	return errs
 }
+
+func validateProxyAddress(hostAddr string) error {
```

Comment:
I couldn't quite see if golang checks for DNS names. Envoy can use either IPv4 or DNS host but not IPv6.